internal/datasourcev2/helpers.go

```diff
@@ -0,0 +1,54 @@
+package datasourcev2
+
+//nolint: godox
```

Comment:
Since this is an evolving bit of work, I'm just silencing my todo notes. I personally think it's completely reasonable to have that in the code as you develop stuff, but if you don't agree, I can pull it out and track it separately.